sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStore.java

```diff
@@ -201,19 +206,29 @@ public Persister getPersister() {
    * must behave as an atomic transaction: On success, everything is written, while on failure
    * nothing is written.
    *
+   * If the number of taskInfo objects to write is greater than MAX_TRANSACTION_SIZE,
+   * taskInfos will be written atomically in groups of MAX_TRANSACTION_SIZE.
+   *
    * @param tasks Tasks to be stored, which each meet the above requirements
    * @throws StateStoreException when persisting TaskInfo information fails, or if its TaskId is
    *                             malformed
    */
   public void storeTasks(Collection<Protos.TaskInfo> tasks) throws StateStoreException {
     Map<String, byte[]> taskBytesMap = new HashMap<>();
-    for (Protos.TaskInfo taskInfo : tasks) {
-      taskBytesMap.put(getTaskInfoPath(namespace, taskInfo.getName()), taskInfo.toByteArray());
-    }
-    try {
-      persister.setMany(taskBytesMap);
-    } catch (PersisterException e) {
-      throw new StateStoreException(e, String.format("Failed to store %d TaskInfos", tasks.size()));
+    List<Protos.TaskInfo> taskInfos = tasks.stream().collect(Collectors.toList());
+    //Only taskInfos are bulk written and require batching, this should only occur at startup when a
+    //new configuration is issued, before the scheduler is initialized and doing work,
+    // so the loss of atomicity from batching should not be noticeable.
+    for (List<Protos.TaskInfo> batchedTaskInfos: Lists.partition(taskInfos, MAX_TRANSACTION_SIZE)) {
```

Comment:
This would still fail if `sizeOf(singeTaskInfo) * 100 > 1 MB` right?
Instead of partitioning by taskInfos, we have to calculate the size of the actual transaction that would be made with ZK. I don't think the approach is foolproof for k8 load testing case.